### PR TITLE
Add an option to skip "npm install" for npm servers

### PR DESCRIPTION
**Pull request by rchl**
Some servers, like the new version of stylelint, bundle all dependencies
and don't need to have any dependencies installed.

This is needed because having no dependencies defined results in
no "node_modules" being created and messes up the logic that detects
if the server is installed or not.

### CODE REVIEW

**Comment by predragnikolic** [comment]
This can also be used for LSP-tailwindcss.

**Comment by rchl** [comment]
And https://github.com/sublimelsp/LSP-leo
@dev-sptg

File: st3/lsp_utils/server_npm_resource.py [hunk 1]
```diff
@@ -71,7 +74,7 @@ def get_status(self) -> int:
 
     def needs_installation(self) -> bool:
         installed = False
-        if path.isdir(path.join(self._server_dest, 'node_modules')):
+        if self._skip_npm_install or path.isdir(path.join(self._server_dest, 'node_modules')):
             # Server already installed. Check if version has changed or last installation did not complete.
             try:
                 src_hash = md5(ResourcePath(self._server_src, 'package.json').read_bytes()).hexdigest()
```
Comment (predragnikolic) [inline st3/lsp_utils/server_npm_resource.py:80]: @rchl do you find it reasonable to throw an human friendly exception if the `self._server_src, 'package.json'` is missing. Something like "Make sure you have a package.json file in the server directory" (or something better that you'll come up with)

Currently, the `server_directory/package.json` file is the only requirement for the NpmHandler to work,
thus throwing an exception for that sounds reasonable for me.

Comment (rchl) [reply st3/lsp_utils/server_npm_resource.py:80]: But the current code has an explicit handle from `FileNotFoundError` exception and re-installs the server in that case. How would throwing an exception play into that?

Comment (rchl) [reply st3/lsp_utils/server_npm_resource.py:80]: We could throw an exception from `install_or_update`, if `package.json` is still missing **after the server is installed** but that would probably only help someone who is authoring the package, not the end users as it shouldn't really be possible for end users to trigger that case.

Comment (predragnikolic) [reply st3/lsp_utils/server_npm_resource.py:80]: >  is still missing after the server is installed but that would probably only help someone who is authoring the package

True, that was my intention, to warn the author. It happened to me when I was developing LSP-tailwindcss, some kind of the error message at that time would be useful. But than again we can leave that as a potential improvement in the future.

Comment (rchl) [reply st3/lsp_utils/server_npm_resource.py]: added code to handle that

**Pull request merged**